ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-ctas-materialized.json

```diff
@@ -205,6 +205,613 @@
+
+        ]},
+        {"query": [
+          {"header":{"schema":"`GRADE` STRING KEY, `NUM_STUDENTS` BIGINT"}},
+          {"row":{"columns":["D",1]}},
+          {"row":{"columns":["F",1]}}
+
+        ]}
+      ]
+    },
+    {
+      "name": "persistent query with COMPLEX filter on all columns + GROUPBY + HAVING + projection +++ pull query table scan and single key lookup",
+      "statements": [
+        "CREATE TABLE INPUT (ID INT PRIMARY KEY, GRADE STRING, RANK INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE NEW_INPUT AS SELECT COUNT(ID) AS NUM_STUDENTS, GRADE FROM INPUT WHERE (ID > 12 AND GRADE IN ('A', 'D')) OR RANK < 2 GROUP BY GRADE HAVING COUNT(ID) > 2;",
+        "SELECT * FROM NEW_INPUT;",
+        "SELECT * FROM NEW_INPUT WHERE NUM_STUDENTS=4;",
+        "SELECT GRADE FROM NEW_INPUT WHERE NUM_STUDENTS=4;"
+
+      ],
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12333, "key": 9, "value": {"GRADE": "F", "RANK": 0}},
+        {"topic": "test_topic", "timestamp": 12346, "key": 11, "value": {"GRADE": "A", "RANK": 1}},
+        {"topic": "test_topic", "timestamp": 12345, "key": 10, "value": {"GRADE": "B", "RANK": 2}},
+        {"topic": "test_topic", "timestamp": 12389, "key": 12, "value": {"GRADE": "C", "RANK": 3}},
+        {"topic": "test_topic", "timestamp": 12397, "key": 14, "value": {"GRADE": "A", "RANK": 4}},
+        {"topic": "test_topic", "timestamp": 12478, "key": 15, "value": {"GRADE": "B", "RANK": 5}},
+        {"topic": "test_topic", "timestamp": 12378, "key": 16, "value": {"GRADE": "A", "RANK": 6}},
+        {"topic": "test_topic", "timestamp": 12378, "key": 18, "value": {"GRADE": "D", "RANK": 7}},
+        {"topic": "test_topic", "timestamp": 12333, "key": 8, "value": {"GRADE": "F", "RANK": -1}},
+        {"topic": "test_topic", "timestamp": 12333, "key": 7, "value": {"GRADE": "F", "RANK": -2}},
+        {"topic": "test_topic", "timestamp": 12333, "key": 6, "value": {"GRADE": "F", "RANK": -3}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`GRADE` STRING KEY, `NUM_STUDENTS` BIGINT"}},
+          {"row":{"columns":["A",3]}},
+          {"row":{"columns":["F",4]}}
+
+        ]},
+        {"query": [
+          {"header":{"schema":"`GRADE` STRING KEY, `NUM_STUDENTS` BIGINT"}},
+          {"row":{"columns":["F",4]}}
+
+        ]},
+        {"query": [
+          {"header":{"schema":"`GRADE` STRING KEY"}},
+          {"row":{"columns":["F"]}}
+
+        ]}
+      ]
+    },
+    {
+      "name": "windowed - select star with HAVING filter",
+      "statements": [
+        "CREATE STREAM INPUT (ID STRING KEY, IGNORED INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE AGGREGATE AS SELECT ID, COUNT(1) AS COUNT FROM INPUT WINDOW TUMBLING(SIZE 1 SECOND) GROUP BY ID HAVING COUNT(1) > 1;",
```

Comment:
@guozhangwang is this the test similar to what you are interested in?

Reply:
Ah yes, thanks!